index.html

```diff
@@ -634,11 +638,20 @@ <h3>The <dfn data-lt="lostpointercapture"><code>lostpointercapture</code> event<
                    <a data-cite="uievents/#event-type-auxclick">auxclick</a> events defined in [[UIEVENTS]] and
-                   [=HTMLElement/contextmenu=] defined in [[HTML]].
-                   The type of these events MUST be <code>PointerEvent</code> but the dispatch process is going to match that of the original specification.
-                   For these events, all <code>PointerEvent</code> specific attributes (defined in this spec) other than <code>pointerId</code> and <code>pointerType</code> will have their default values.
-                   The <code>pointerId</code> and <code>pointerType</code> of these events MUST be the same as the PointerEvents that caused these events if they are generated by a pointing device.
-                   If the events are generated by a non-pointing device (such as voice recognition software or a keyboard interaction), <code>pointerId</code> MUST be <code>-1</code> and <code>pointerType</code> MUST be an empty string.</p>
+                   [=HTMLElement/contextmenu=] defined in [[HTML]]. The type of these events MUST be <code>PointerEvent</code>,
+                   but the dispatch process is going to match that of the original specification.</p>
+                   For these events, all <code>PointerEvent</code> specific attributes (defined in this spec) other than <code>pointerId</code> and <code>pointerType</code> will have their default values. In addition:</p>
+                   <ul>
+                       <li>If the events are generated by a pointing device, their <code>pointerId</code> and <code>pointerType</code> MUST be the same as the PointerEvents that caused these events.</li>
+                       <li>If the events are generated by a non-pointing device (such as voice recognition software or a keyboard interaction), <code>pointerId</code> MUST be <code>-1</code> and <code>pointerType</code> MUST be an empty string.</li>
+                    </ul>
+                <p>As noted in {{PointerEvent}}, [[[CSSOM-VIEW]]] proposes to redefine the various coordinate properties (<code>screenX</code>, <code>screenY</code>, <code>pageX</code>, <code>pageY</code>, <code>clientX</code>,
+                   <code>clientY</code>, <code>x</code>, <code>y</code>, <code>offsetX</code>, <code>offsetY</code>) as <code>double</code>, to allow for fractional coordinates.
+                   However, this change — when applied only to {{PointerEvent}}, but not to regular {{MouseEvent}} — has proven to lead to web compatibility issues with legacy code
+                   in the case of <code>click</code>, <code>auxclick</code>, and <code>contextmenu</code>. For this reason, user agents that have only implemented the proposed
+                   change in [[[CSSOM-VIEW]]] for {{PointerEvent}} MUST round the various coordinate properties for the <code>click</code>, <code>auxclick</code>, and <code>contextmenu</code>
```

Comment:
*Rounding* isn't necessarily going to make the pointerevent equivalent to the mouse event. E.g. In chromium, we [floor the coordinates](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/events/mouse_event.h;l=140). Is there some wording we can use here to say they must be equivalent to the compatibility mouse events?

Reply:
for compat mouse events, we don't say anything regarding coordinates at all at the moment. and generating compat mouse events is an optional feature that browser may or may not choose to implement. this may lead to a further conditional sentence here "if the user agent implements compatibility mouse events, do the same as there. otherwise..." and then that second case would still need to be explained. hmm

at the end of 4.1 we have a note with

> When the user agent calculates tiltX / tiltY from azimuthAngle / altitudeAngle it SHOULD round the final integer values using Math.round [ECMASCRIPT] rules.

maybe here we can explicitly say it should be Math.floor ?

Reply:
i can already see us having to now also say something in the compat mouse events section about coordinates... got ourselves a bit tangled here...

Reply:
It occurred to me after writing this that the mapping from floating point to integer is entirely developer visible ([can observe both the PointerEvent and the MouseEvent](https://jsbin.com/sagenev/edit?html,js,output)), so it probably should be well defined which way numbers are rounded. As far as I can tell from that jsbin chromium is the odd browser out to even use floating point coordinates for PointerEvents.

Reply:
wondering if the WebIDL for PE can/should be changed to normatively define coordinates as `double` (assume it would be easier for PE to do this, as it's a new type of event, rather than what CSSOM tried to do with redefining existing MouseEvent

Reply:
...or if this is just something that we just leave as is (but which then leads to having these awkward "if you're following CSSOM, then..." type constructs)